components/eventing-controller/pkg/env/backend_config.go

```diff
@@ -18,19 +18,26 @@ type BackendConfig struct {
 }
 
 type PublisherConfig struct {
-	Image                  string `envconfig:"PUBLISHER_IMAGE" default:"eu.gcr.io/kyma-project/event-publisher-proxy:c06eb4fc"`
-	ImagePullPolicy        string `envconfig:"PUBLISHER_IMAGE_PULL_POLICY" default:"IfNotPresent"`
-	PortNum                int    `envconfig:"PUBLISHER_PORT_NUM" default:"8080"`
-	MetricsPortNum         int    `envconfig:"PUBLISHER_METRICS_PORT_NUM" default:"8080"`
-	ServiceAccount         string `envconfig:"PUBLISHER_SERVICE_ACCOUNT" default:"eventing-publisher-proxy"`
-	Replicas               int32  `envconfig:"PUBLISHER_REPLICAS" default:"1"`
-	RequestsCPU            string `envconfig:"PUBLISHER_REQUESTS_CPU" default:"32m"`
-	RequestsMemory         string `envconfig:"PUBLISHER_REQUESTS_MEMORY" default:"64Mi"`
-	RequestTimeout         string `envconfig:"PUBLISHER_REQUEST_TIMEOUT" default:"5s"`
-	LimitsCPU              string `envconfig:"PUBLISHER_LIMITS_CPU" default:"100m"`
-	LimitsMemory           string `envconfig:"PUBLISHER_LIMITS_MEMORY" default:"128Mi"`
-	EnableJetStreamBackend bool   `envconfig:"ENABLE_JETSTREAM_BACKEND" default:"false"`
-	PriorityClassName      string `envconfig:"PUBLISHER_PRIORITY_CLASS_NAME" default:""`
+	Image             string `envconfig:"PUBLISHER_IMAGE" default:"eu.gcr.io/kyma-project/event-publisher-proxy:c06eb4fc"`
+	ImagePullPolicy   string `envconfig:"PUBLISHER_IMAGE_PULL_POLICY" default:"IfNotPresent"`
+	PortNum           int    `envconfig:"PUBLISHER_PORT_NUM" default:"8080"`
+	MetricsPortNum    int    `envconfig:"PUBLISHER_METRICS_PORT_NUM" default:"8080"`
+	ServiceAccount    string `envconfig:"PUBLISHER_SERVICE_ACCOUNT" default:"eventing-publisher-proxy"`
+	Replicas          int32  `envconfig:"PUBLISHER_REPLICAS" default:"1"`
+	RequestsCPU       string `envconfig:"PUBLISHER_REQUESTS_CPU" default:"32m"`
+	RequestsMemory    string `envconfig:"PUBLISHER_REQUESTS_MEMORY" default:"64Mi"`
+	RequestTimeout    string `envconfig:"PUBLISHER_REQUEST_TIMEOUT" default:"5s"`
+	LimitsCPU         string `envconfig:"PUBLISHER_LIMITS_CPU" default:"100m"`
+	LimitsMemory      string `envconfig:"PUBLISHER_LIMITS_MEMORY" default:"128Mi"`
+	PriorityClassName string `envconfig:"PUBLISHER_PRIORITY_CLASS_NAME" default:""`
+
+	// jetstream
+	EnableJetStreamBackend  bool   `envconfig:"ENABLE_JETSTREAM_BACKEND" default:"false"`
+	JSStreamName            string `envconfig:"JS_STREAM_NAME" default:"kyma"`
+	JSStreamStorageType     string `envconfig:"JS_STREAM_STORAGE_TYPE" default:"memory"`
+	JSStreamRetentionPolicy string `envconfig:"JS_STREAM_RETENTION_POLICY" default:"interest"`
+	JSStreamMaxMessages     int64  `envconfig:"JS_STREAM_MAX_MSGS" default:"-1"`
+	JSStreamMaxBytes        int64  `envconfig:"JS_STREAM_MAX_BYTES" default:"-1"`
```

Comment:
What is the purpose of passing the extra fields: `JSStreamStorageType`, `JSStreamRetentionPolicy`, `JSStreamMaxMessages` and `JSStreamMaxBytes` to EPP? Is it needed in EPP to initialise the jetStream connection or anyplace else?

Reply:
I pass them in order to match the jetstream config fully in the EPP when publishing the events. There I will not only check for the jetstreamName, but also for the storageType, RetentionPolicy and other properties.

Reply:
I checked once again, seems that stream name is unique for jetstream, so we need to pass only the stream name